docs/compiler-exceptions.rst

```diff
@@ -0,0 +1,275 @@
+.. index:: compiling, compiler, InvalidTypeException, common, exceptions, StructureException, ConstancyViolationException, NonPayableViolationException, InvalidLiteralException, TypeMismatchException, EventDeclarationException, VersionException, SyntaxException, ArrayIndexException, ZeroDivisionException, EvmVersionException, CompilerPanic, VariableDeclarationException
+
+Compiler Exceptions
+*******************
+
+.. _exceptions-common::
+
+Common Compiler Exceptions
+==========================
+
+These are examples of the more common compiling exception errors that you could see when
+compiling a ``vyper`` file for deployment on the Ethereum Virtual Machine.
+
+
+.. py:exception:: ArrayIndexException
+
+This exception will occur when an invalid index number of an array is referenced.  The terminal will point out the line 
```

Comment:
The usual style is to indent text and examples beneath each directive. Take a look at [built-in-functions.rst](https://github.com/vyperlang/vyper/blob/master/docs/built-in-functions.rst) for an example.

Reply:
Fixed